imagersite/imager_images/views.py

```diff
@@ -1,71 +1,139 @@
-from django.shortcuts import render, redirect, get_object_or_404
+from django.shortcuts import redirect, get_object_or_404
 from .models import Album, Photo
 from django.http import Http404
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+from django.views.generic import CreateView
+from .forms import AlbumForm, PhotoForm
 
 
-def photo_view(request, photo_id):
-    photo = get_object_or_404(Photo, id=photo_id)
+class PhotoView(DetailView):
+    template_name = 'imager_images/photo.html'
+    model = Photo
+    pk_url_kwarg = 'photo_id'
 
-    username = request.user.get_username()
+    def get(self, *args, **kwargs):
+        try:
+            self.photo = get_object_or_404(Photo, id=kwargs['photo_id'])
+        except KeyError:
+            raise Http404
+        self.username = self.request.user.get_username()
 
-    context = {}
-    if photo.user.username != username and photo.published != 'PUBLIC':
-        raise Http404('Photo not found.')
+        if self.photo.user.username != \
+                self.username and self.photo.published != 'PUBLIC':
+            raise Http404('Photo not found.')
 
-    if photo.user.username == username or photo.published == 'PUBLIC':
-        context['photo'] = photo
+        return super().get(*args, **kwargs)
 
-    return render(request, 'imager_images/photo.html', context)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['photo'] = self.photo
```

Comment:
Just specify the `context_object_name` to be `'photo'` and the `Photo` found by the `.get_object()` method with be stored in the context under the `'photo'` key. Then you do not need to overwrite `.get_context_data()`.